internal/featuretests/kubernetes.go

```
@@ -32,77 +35,74 @@ func IngressBackend(svc *v1.Service) *networking_v1.IngressBackend {
-
-	CRL = `-----BEGIN X509 CRL-----
-MIHlMIGMAgEBMAoGCCqGSM49BAMCMBsxGTAXBgNVBAMTEGNsaWVudC1yb290LWNh
-LTEXDTIyMDYyMTA5NDQ0NVoXDTIyMDYyODA5NDQ0NVowGzAZAggW+pmWu/XnExcN
-MjIwNjIxMDk0NDQ1WqAjMCEwHwYDVR0jBBgwFoAULRlmjBtfjbzwV2WeO9Vj5pWO
-h5gwCgYIKoZIzj0EAwIDSAAwRQIhANVFCqByuASAcbz6ovyvi5KCtPfNjHjxVaNT
-x69LFPN1AiA5pF5rqHy1FBctZBTW+3LTEEX35j3p1++zcNu8oHMO/w==
------END X509 CRL-----`
-)
+var CACertificate = certyaml.Certificate{
+	Subject: "CN=ca",
+}
+
+var ServerCertificate = certyaml.Certificate{
+	Issuer:          &CACertificate,
+	Subject:         "CN=www.example.com",
+	SubjectAltNames: []string{"DNS:www.example.com"},
+}
+
+var ClientCertificate = certyaml.Certificate{
+	Issuer:  &CACertificate,
+	Subject: "CN=client",
+}
 
-func Secretdata(cert, key string) map[string][]byte {
-	return map[string][]byte{
-		v1.TLSCertKey:       []byte(cert),
-		v1.TLSPrivateKeyKey: []byte(key),
+var CRL = certyaml.CRL{
+	Issuer: &CACertificate,
+}
+
+func TLSSecret(name string, credential *certyaml.Certificate) *v1.Secret {
+	cert, key, err := credential.PEM()
+	if err != nil {
+		panic(err)
+	}
+	return &v1.Secret{
+		ObjectMeta: fixture.ObjectMeta(name),
+		Type:       v1.SecretTypeTLS,
+		Data: map[string][]byte{
+			v1.TLSCertKey:       cert,
+			v1.TLSPrivateKeyKey: key,
+		},
 	}
 }
 
+func CASecret(name string, credential *certyaml.Certificate) *v1.Secret {
+	cert, _, err := credential.PEM()
+	if err != nil {
+		panic(err)
+	}
+	return &v1.Secret{
+		ObjectMeta: fixture.ObjectMeta(name),
+		Type:       v1.SecretTypeOpaque,
+		Data: map[string][]byte{
+			dag.CACertificateKey: cert,
+		},
+	}
+}
+
+func CRLSecret(name string, credential *certyaml.CRL) *v1.Secret {
+	crl, err := credential.PEM()
+	if err != nil {
+		panic(err)
+	}
+
+	return &v1.Secret{
+		ObjectMeta: fixture.ObjectMeta(name),
+		Type:       v1.SecretTypeOpaque,
+		Data: map[string][]byte{
+			dag.CRLKey: crl,
+		},
+	}
+}
+
+func PEMBytes(cert *certyaml.Certificate) []byte {
+	c, _, _ := cert.PEM()
```

here we're ignoring the error, but in other places we use a `panic`

should we either ignore the error everywhere or maybe make these test helpers and pass `testing.T` so we can make an assertion there is no error?

---

Done!